### PR TITLE
sync with the internal sdk

### PR DESCRIPTION
**Pull request by jingweicb**
Fixes # .
check datat test
Screenshot 2022-12-22 at 11 28 00 AM

check construction
Screenshot 2022-12-22 at 11 28 16 AM

### CODE REVIEW

File: client/types.go [hunk 1]
```diff
@@ -134,7 +134,6 @@ type RPCTransaction struct {
 }
 
 type RosettaTxReceipt struct {
-	Type           uint8  `json:"type,omitempty"`
```
Comment (shrimalmadhur) [inline client/types.go:137]: This cannot be removed IMO. This is added by @xiaying-peng for https://github.com/mdehoog/op-rosetta

File: services/mapper.go [hunk 2]
```diff
@@ -84,15 +156,9 @@ func FeeOps(tx *evmClient.LoadedTransaction) []*RosettaTypes.Operation {
...
-	// in internal is 
-	// burntOp := &RosettaTypes.Operation{
-	// 	OperationIdentifier: &RosettaTypes.OperationIdentifier{
-	// 		Index: 0, // nolint:gomnd
-	// 	},
-	idx := len(ops)
 	burntOp := &RosettaTypes.Operation{
 		OperationIdentifier: &RosettaTypes.OperationIdentifier{
-			Index: int64(idx), // nolint:gomnd
+			Index: 0, // nolint:gomnd
```
Comment (shrimalmadhur) [inline services/mapper.go:161]: @ash-krishnan @xiaying-peng to confirm this change

File: services/mapper.go [hunk 3]
```diff
@@ -271,13 +334,8 @@ func TraceOps(
 	return ops
 }
 
-// Erc20Ops returns a list of erc20 operations parsed from the log from a transaction receipt
-func Erc20Ops(
```
Comment (shrimalmadhur) [inline services/mapper.go:275]: I think we need to leave this function in this format. please revert the syntax.

File: services/mapper.go [hunk 4]
```diff
@@ -31,7 +31,79 @@ const (
...
+				Amount: evmClient.Amount(transfer.Value, sdkTypes.Currency),
+			}
+		ops = append(ops, doubleOp)
+		count++
+		}
+	}
+	return ops
+}
+
+func TransferOps(tx *evmClient.LoadedTransaction, startIndex int) []*RosettaTypes.Operation {
```
Comment (shrimalmadhur) [inline services/mapper.go:95]: Add a comment line for this. We should add a linter ideally to this repo which can detect this.

File: services/construction/preprocess.go [hunk 5]
```diff
@@ -97,62 +91,41 @@ func (s *APIService) ConstructionPreprocess( //nolint
...
-				fmt.Errorf("%s is not a valid gas limit", v),
-			)
+			return nil, sdkTypes.WrapErr(sdkTypes.ErrInvalidInput, fmt.Errorf("%s is not a valid gas limit", v))
 		}
 		preprocessOptions.GasLimit = bigObj
 	}
 	if v, ok := req.Metadata["nonce"]; ok {
 		stringObj, ok := v.(string)
 		if !ok {
-			return nil, sdkTypes.WrapErr(
```
Comment (shrimalmadhur) [inline services/construction/preprocess.go:134]: I think this format is more readable for errors so let s keep it?

**Review by shrimalmadhur — CHANGES_REQUESTED** [review]
Almost all the formatting changes need not to be done. They are more readable and this change is making it less readable. Let's revert. 

Few changes needs to be confirmed by @xiaying-peng @ash-krishnan  so let's wait.

File: services/mapper.go (hunk 2 above)
Comment (jingweicb) [reply services/mapper.go:164]: this is related to support whitelist, @ash-krishnan will have more context

File: services/block_service_test.go [hunk 6]
```diff
@@ -40,9 +40,24 @@ import (
 )
 
 const (
-	hsh = "0xd83b1dcf7d47c4115d78ce0361587604e8157591b118bd64ada02e86c9d5ca7e"
+  hsh = "0xd83b1dcf7d47c4115d78ce0361587604e8157591b118bd64ada02e86c9d5ca7e"
 )
 
+// nolint:unused
+func loadTransfers(transferFile string) []*client.EVMTransfer {
```
Comment (ash-krishnan) [inline services/block_service_test.go:47]: sorry left this function in the generic-sdk by mistake, but we don't need it or use it, so you can just get rid of it

**Comment by cb-heimdall** [comment]
Review Error for ash-krishnan @ 2023-01-06 19:05:14 UTC
User must have write permissions to review

**Review by xiaying-peng — APPROVED** [review]

**Pull request merged**